### PR TITLE
Add v1 apiVersion to CRDs

### PR DESCRIPTION
**Pull request by aharonYK**
# Description

Change Component/Configuration/Subscription CRDs apiVersion to v1 #6582
let me know if this is what you meant. it was a very good experience working on this i didn't came across it before
## Issue reference
#6582

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by aharonYK** [comment]
please let me know if it need fixing

File: charts/dapr/crds/subscription.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ spec:
       - v1
       - v2alpha1
   versions:
-  - name: v1alpha1
+  - name: v1
```
Comment (yaron2) [inline charts/dapr/crds/subscription.yaml:24]: For Subscriptions, we need to add `v1` with the same schema as `v2alpha1` and leave `v1alpha` unchanged.

**Review by yaron2 — CHANGES_REQUESTED** [review]

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
These changes need to happen in the `pkg/api` structs, which are used to generate the CRDs with `make code-generate`. Also, please do not delete old versions because that's a backwards-incompatible change. We need to continue supporting old versions and use a webhook to convert to `v1`.

**Comment by daixiang0** [comment]
@aharonYK please resolve the comments :)

File: charts/dapr/crds/subscription.yaml (hunk 1 above)
Comment (berndverst) [reply charts/dapr/crds/subscription.yaml:24]: And we also need to keep `v2alpha1` around because that is also in use.

**Comment by cicoyle** [comment]
Hey @aharonYK - Will you be able to pick this back up?

**Pull request closed without merging**